### DISCUSSION TITLE
Timepoint matching

### DISCUSSION THREAD

**Discussion by mwalenta1** [cnr-isti-vclab/TagLab#108] [Q&A]
Hello community

We are monitoring benthic changes in the Caribbean. We have several timepoints and are looking for the program to use original timepoints to then automatically match species and growth/recession of colonies.

Currently, the program is not linking second timeline matches. We were able to add a second time point and able to go to split view but when we tried to "compute automatic matches" they are not ID'd in the second time point. Let us know if there are more detailed instructions or if someone is able to zoom about this. 

Thank you for any help!!

**Comment by Joetown94** [comment]
Hello,
      I see this comment is still open, I am currently working through the same issue and was wondering if you had any success.

From my investigation we initially had some trouble as the images weren't perfectly aligned to one another, and had to be aligned (using the alignment tool) and then scaled to appropriate pixel sizing as well. However despite doing both of these steps, the automatic matches tool still won't identify any matches.

In trying to simply export the label image and re import into the new image, despite full alignment of the two images, the segments are placed offset, likely because of differences in pixel size and dimensions of the mosaic images.

From here, I'm fairly lost as to what might be the next steps for this tool to work, or if instead it's more efficient to simply annotate both images manually.

Any suggestions or assistance appreciated.

**Reply by maxcorsini** [reply]
Hi Joe,

since you wrote this: 

"From here, I'm fairly lost as to what might be the next steps for this tool to work, or if instead it's more efficient to simply annotate both images manually."

I would like to clarify that the "automatic matches tool" matches the annotations. So, you need to annotate both of the timepoints. 

Have you annotated both ? If yes, you did the right thing to use the alignment tool, because the automatic matches tool needs that the orthoimage should be coregistered. To check this you can split the view and inspect if the exactly the same locations are shown in the let and right view (with the same scale).

Let me know if this helps you.

Best,
Massimiliano

**Comment by Joetown94** [comment]
Thank you for the clarification and response, I had suspected this may be the case when I originally explored using this tool. That makes sense.

I have not yet fully annotated both images, as I'm still exploring the most efficient pipeline for doing fully species-specific semantic segmentation of a mosaic image through several timepoints. Do you recommend independently segmenting each image as the most efficient workflow for this, or another workflow or tool to consider for this purpose?

Thanks,
Joe

**Reply by maxcorsini** [reply]
Hi Joe,

it depends on your data. If you have many plots, my suggestion is to develop an automatic classifiers using a part of them, automatically segment the others (to save time), make the necessary corrections (the AI is not perfect) and then track the changes in the different time points. If you have few maps, you can directly annotate all using the semi-automatic tools available in Taglab to speed up the annotation.

If you need assistance for the development of the automatic classifier feel free to contact us for suggestions specific to your data (we can set up a call). 

Best,
Massimiliano
